src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2247,14 +2706,56 @@ class CgroupUtils(object):
             return False
         return True
 
-    def add_pids(self, pids, jobid):
+    def _get_pids_in_sid(self, sid=None):
+        """
+        Return a list of all PIDS associated with a session ID
+        """
+        pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Method called" % caller_name())
+        pids = []
+        if not sid:
+            return pids
+        # Older kernels will not have a task directory
+        if os.path.isdir(os.path.join(os.sep, 'proc', 'self', 'task')):
+            check_tasks = 1
+        else:
+            check_tasks = 0
+        for file in glob.glob(os.path.join(os.sep, 'proc', '[0-9]*', 'stat')):
+            try:
+                with open(file, 'r') as fd:
+                    entries = fd.readline().split(' ')
+                    if int(entries[5]) != sid:
+                        continue
+                    if check_tasks:
+                        # Thread group leader will have a task entry.
+                        # No need to append entry from stat file.
+                        taskdir = os.path.join(os.path.dirname(file), 'task')
+                        for task in glob.glob(os.path.join(taskdir, '[0-9]*')):
+                            pid = int(os.path.basename(task))
+                            if pid not in pids:
+                                pids.append(pid)
+                    else:
+                        # Append entry from stat file
+                        if int(entries[0]) not in pids:
+                            pids.append(int(entries[0]))
+            except:
+                raise
```

Comment:
I added a comment and changed the raise to a pass. We're inspecting proc, which could change while we're reading files. That should not cause a fatal error since it is expected (occasionally). When I pasted the method I had unit tested into the hook I didn't update that piece.